app/api/domains/cho.py

```diff
@@ -84,20 +84,100 @@
 @router.get("/")
 async def bancho_http_handler():
     """Handle a request from a web browser."""
+    new_line = "\n"
+    matches = list(
+        filter(lambda match: isinstance(match, Match), app.state.sessions.matches),
+    )
+    players = [p for p in app.state.sessions.players if not p.bot_client]
+
     packets = app.state.packets["all"]
+    return HTMLResponse(
+        f"""
+<!DOCTYPE html>
+<body style="font-family: monospace; white-space: pre-wrap;">Running bancho.py v{app.settings.VERSION}
+
+<a href="online">{len(players)} online players</a>
+<a href="matches">{len(matches)} matches</a>
+
+<b>packets handled ({len(packets)})</b>
+{new_line.join([f"{packet.name} ({packet.value})" for packet in packets])}
+
+<a href="https://github.com/osuAkatsuki/bancho.py">Source code</a>
+</body>
+</html>""",
+    )
+
+
+@router.get("/online")
+async def bancho_list_user():
+    """see who's online"""
+    new_line = "\n"
+    players = list(filter(lambda p: not p.bot_client, app.state.sessions.players))
+    bots = list(filter(lambda p: p.bot_client, app.state.sessions.players))
+    id_max_length = len(str(max(p.id for p in app.state.sessions.players)))
 
     return HTMLResponse(
-        b"<!DOCTYPE html>"
-        + "<br>".join(
-            (
-                f"Running bancho.py v{app.settings.VERSION}",
-                f"Players online: {len(app.state.sessions.players) - 1}",
-                '<a href="https://github.com/osuAkatsuki/bancho.py">Source code</a>',
-                "",
-                f"<b>packets handled ({len(packets)})</b>",
-                "<br>".join([f"{packet.name} ({packet.value})" for packet in packets]),
-            ),
-        ).encode(),
+        f"""
+<!DOCTYPE html>
+<body style="font-family: monospace;  white-space: pre-wrap;"><a href="/">back</a>
+users:
+{new_line.join(
+    map(
+        lambda p: f"({str(p.id).rjust(id_max_length)}): {p.safe_name}",
```

Comment:
if i remember correctly, i think this should work as well for the `f"{str(x).rjust(y)}"` cases, but i don't particularly care either way
```suggestion
        lambda p: f"({p.id:>{id_max_length}}): {p.safe_name}",
```

Reply:
LGTM

Reply:
how to do a ljust? <: ?

Reply:
(there are many more cases of this though - my only request is to pick one style and use it consistently)

Reply:
almost - `:<`

Reply:
in python `:` is the delimiter between the data and the format spec

e.g.
```py
balance = 123.4567
print(f"Balance: ${123:.2f}") # here, the format spec is `.2f`, for "2 decimal float"
> 123.45
```

Reply:
more format specs here https://www.w3schools.com/python/ref_string_format.asp

Reply:
updated